File: backend/components/metiers/administration_publique/views/router/test_router.js
// test_router.js – Test du routeur Express pour Threed
const request = require('supertest');
const express = require('express');
const router = require('../router');

describe('Router 3D', () => {
  let app;
  beforeAll(() => {
    app = express();
    app.use(express.json());
    app.use(router);
  });

  it('GET /3d retourne la liste des d3s', async () => {
    const res = await request(app).get('/3d');
    expect(res.statusCode).toBe(200);
    expect(res.body).toHaveProperty('d3s');
    expect(res.body).toHaveProperty('total');
  });

  it('POST /3d crée un d3', async () => {
    const data = { nom: 'Test 3D', description: 'desc', type: 'objet' };
    const res = await request(app).post('/3d').send(data);
    expect([200, 201]).toContain(res.statusCode);
    expect(res.body.nom).toBe('Test 3D');
  });
});

File: backend/components/metiers/administration_publique/views/router/test_router.py
# test_router.py – Test du routeur FastAPI pour Threed
from fastapi.testclient import TestClient
from backend.components.metiers.threed.views.router import router
from fastapi import FastAPI

app = FastAPI()
app.include_router(router)
client = TestClient(app)

def test_list_d3s():
    response = client.get("/3d")
    assert response.status_code == 200
    assert "d3s" in response.json()
    assert "total" in response.json()

def test_create_d3():
    data = {"nom": "Test 3D", "description": "desc", "type": "objet"}
    response = client.post("/3d", json=data)
    assert response.status_code == 201 or response.status_code == 200
    assert response.json()["nom"] == "Test 3D"
